core/services/ocr2/plugins/promwrapper/plugin.go

```diff
@@ -343,3 +352,13 @@ func (p *promPlugin) Close() error {
 
 	return p.wrapped.Close()
 }
+
+func (p *promPlugin) setEndTime(timestamp types.ReportTimestamp, cache *cache.Cache) {
+	cache.SetDefault(timestampToKey(timestamp), time.Now().UTC())
+}
+
+func timestampToKey(timestamp types.ReportTimestamp) string {
+	jsonData, _ := json.Marshal(timestamp)
```

Comment:
Unchecked error.

The linter should have flagged this :thinking:

Reply:
Not sure we need a JSON marshal, I think the report timestamp can be safely marshaled to a string without any errors:

```golang
key := fmt.Sprintf("%x_%d_%d", timestamp.ConfigDigest[:], timestamp.Epoch, timestamp.Round)
```

Reply:
Agree with @makramkd, no need to run json marshaller, probably `sprintf` will be faster and more than enough here